migtools/oadp-non-admin
# Facilitate deletion of VeleroBackup object via NAB object #58

Fix for deletion of NonAdminBackup object and corresponding VeleroBackup with updated logic to Status not being reconciled every time Status is being changed.

Please check the [docs/design/nab_and_nar_status_update.md diagram](https://github.com/mpryc/oadp-non-admin/blob/velero_delete_event_new/docs/design/nab_and_nar_status_update.md#status-update-scenarios) for the full workflow of the NAB and VeleroBackup deletion and updated Status and Conditions reconciles.
 In few simple steps:
 - The Finalizer is set on NAB object just before VeleroBackup is created
 - The deletion of NAB, can be triggered in three ways:
  1) By setting NAB spec.deleteBackup 
  2) By setting NAB spec.forceDeleteBackup
  3) By running `oc delete` or other k8s API server deletion call such as namespace deletion

## Why the changes were made

To fix #58 

## How to test the changes made
Implemented additional logic within reconcile loop.

This PR replaces #105 

## Testing
1. Deploy OADP with NAC
2. Create NAB Object as in other cases (DPA should also be already in the cluster)
3. Wait for the Velero Backup object to be created
  3a. Run `oc delete <nab>`
        Expected result: NAB not deleted, condition set to inform why it has not been deleted
  3b. With perform steps 1-3a and then set the spec.forceDeleteBackup to true
        Expected result: NAB should be removed, Velero Backup should be removed, S3 Backup should not be removed.
  3c. Perform steps 1-2 and then set the NAB spec.deleteBackup 
        Expected result: NAB should be removed, Velero Backup should be removed, S3 Backup should be removed
